pkg/repo/repo.go

```diff
@@ -302,3 +303,55 @@ func UpdateRoleKeys(repo *tuf.Repo, store tuf.LocalStore, role string, keys []*d
 	}
 	return nil
 }
+
+func MarshalMetadata(v interface{}) ([]byte, error) {
+	// We don't need to canonically encode the payload in the store.
+	b, err := json.Marshal(v)
+	if err != nil {
+		return nil, err
+	}
+
+	var out bytes.Buffer
+	if err := json.Indent(&out, b, "", "\t"); err != nil {
+		return nil, err
+	}
+
+	return out.Bytes(), nil
+}
+
+func SetSignedMeta(store tuf.LocalStore, role string, s *data.Signed) error {
+	b, err := MarshalMetadata(s)
+	if err != nil {
+		return err
+	}
+	return store.SetMeta(role, b)
+}
+
+// BumpMetadataVersion increments the version of the manifest.
+// This ONLY handles targets types! The repo.SetRootVersion, repo.SetTargetsVersion,
+// or repo.SetSnapshotVersion, or repo.SetTimestampVersion handle top-level
+// metadata.
```

Comment:
```suggestion
// BumpMetadataVersion increments the version of the manifest.
// This ONLY handles delegated targets roles! The repo.SetRootVersion,
// repo.SetTargetsVersion, or repo.SetSnapshotVersion, or
// repo.SetTimestampVersion handle top-level metadata.
```

We should provide better API for this in go-tuf :-|

Reply:
Filed an issue in go-tuf https://github.com/theupdateframework/go-tuf/issues/406